src/main/resources/assets/js/ArchiveRestoreDialog.ts

```diff
@@ -45,11 +46,36 @@ export class ArchiveRestoreDialog
     }
 
     protected doAction() {
-        new RestoreArchivedRequest(this.items.map(item => item.getId())).sendAndParse().then((taskId: TaskId) => {
+        new RestoreArchivedRequest(this.getItemsToRestore()).sendAndParse().then((taskId: TaskId) => {
             this.progressManager.pollTask(taskId);
         }).catch(DefaultErrorHandler.handle);
     }
 
+    private getItemsToRestore(): string[] {
+        const itemsToDelete: ArchiveViewItem[] = [];
+
+        this.items.forEach((item: ArchiveViewItem) => {
+            const contains: boolean = itemsToDelete.some((itemToDelete: ArchiveViewItem, index: number) => {
+                if (item.getData().getPath().isDescendantOf(itemToDelete.getData().getPath())) {
```

Comment:
This and the next condition should be extracted into a separate static method called `isDescendantOf` which checks if one item is a descendant of another.

We also have exactly the same check (albeit for other classes) [here](https://github.com/enonic/app-contentstudio-plus/blob/master/src/main/resources/assets/js/ArchiveTreeGrid.ts#L79), so maybe think about how to have a unified check.

Reply:
Updated